source/slang/slang-emit-c-like.cpp

```diff
@@ -3526,6 +3546,11 @@ void CLikeSourceEmitter::emitFuncDecorationsImpl(IRFunc* func)
 
 void CLikeSourceEmitter::emitStruct(IRStructType* structType)
 {
+    if (shouldEmitOnlyHeader() && !structType->findDecoration<IRExternCppDecoration>())
```

Comment:
Instead of skipping here, we should just make sure we never generate the work to emit the struct from the beginning. See `CLikeSourceEmitter::computeEmitActions` where we are trying to figure out what insts needs to be emitted. We should just make sure the irrelavent structs are not added. Also if you have:
```
struct MyType {}

[HeaderExport]
void myFunc(MyType t);
```

That should result `MyType` to be emitted automatically. By carefully handling `computeEmitActions` we should be able to handle this gracefully.